packages/dds/tree/src/edit-manager/editManager.ts

```diff
@@ -0,0 +1,163 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ChangeFamily } from "../change-family";
+import { AnchorSet, Delta } from "../tree";
+
+export interface Commit<TChangeset> {
+    sessionId: SessionId;
+    seqNumber: SeqNumber;
+    refNumber: SeqNumber;
+    changeset: TChangeset;
+}
+
+type SessionId = number;
+type SeqNumber = number;
+
+/**
+ * Represents a local branch of a document and interprets the effect on the document of adding sequenced changes,
+ * which were based on a given session's branch, to the document history
+ */
+// TODO: Remove commits when they are no longer in the collab window
+// TODO: Try to reduce this to a single type parameter
+// TODO: Move logic into Rebaser if possible
+export class EditManager<TChangeset, TChangeFamily extends ChangeFamily<any, TChangeset>> {
+    private readonly trunk: Commit<TChangeset>[] = [];
+    private readonly branches: Map<SessionId, Branch<TChangeset>> = new Map();
+    private readonly localChanges: TChangeset[] = [];
+
+    public constructor(private readonly localSessionId: SessionId,
+        private readonly changeFamily: TChangeFamily,
+    ) { }
+
+    public addSequencedChange(newCommit: Commit<TChangeset>, anchors?: AnchorSet): Delta.Root {
+        if (newCommit.sessionId === this.localSessionId) {
```

Comment:
we should probably assert that the change matches the oldest local change. If nothing else that would help document whats happening here.

Reply:
I don't think there's an easy way to do that right now, but I'll add a comment explaining the logic.